### PR TITLE
feat: Implemented Trello Piece

### LINKED ISSUE #278: Trello Piece
- Actions:
    - Add Card.
- Triggers:
    - New Card.

### PR DESCRIPTION
**Pull request by ShayPunter**
## What does this PR do?

Implements a 'create card' action piece for Trello. 
Implements a 'create card' action piece for Trello
Implements documentation update effecting:
- Apps.mdx
- [new] trello.mdx
- quickstart piece guide (updated to reflect new import paths)

Fixes #278 (partial fix)

## Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## How should this be tested?

Tested in a codespaces environment, please use the following test flow outlined in the documentation:

Obtain API Key & Token:
1. Go to https://trello.com/app-key
2. Copy 'personal key' and enter it into the Trello API Key connection
3. Click 'generate a token' in trello
4. Copy the token and enter it into the Trello Token connection
5. Your connection should now work!

Creating your first Trello card using Activepiece:
1. Create a new collection
2. Select trigger as 'schedule' and we will use its default value of 5 minutes
3. Click on + to add a new piece
4. Select the 'Trello' piece
5. Select the 'Create Card' Action on the right-hand sidebar
6. Select your Trello API Key connection you created above
7. Select your Trello Token Key connection above
8. Head over to the Trello board you wish to create a new card in
9. Create a new card, lets call it 'test'
10. Click into that card
11. In the URL, add '.json', this should return a wall of JSON text
12. Look for a value called 'idList', it should look something like this: `"idList":"5aea17e911c71fee96375d94"`
13. Copy the idList value, in our case `5aea17e911c71fee96375d94` and enter it into the idList field
14. Set the card name you'd like to create
15. Set a description to enter into the card (optional)
16. Lastly, lets test our flow by clicking 'Test Flow'
17. A new card should have been created in the selected list
18. Now click publish, and thats it!

Fetching a card via Activepieces

1. Create a new collection
2. Select trigger as 'schedule' and we will use its default value of 5 minutes
3. Click on + to add a new piece
4. Select the 'Trello' piece
5. Select the 'Get Card' Action on the right-hand sidebar
6. Select your Trello API Key connection you created above
7. Select your Trello Token Key connection above
8. Head over to the Trello board you wish to create a new card in
9. Create a new card, lets call it 'test'
10. Click into that card
11. In the URL, add '.json', this should return a wall of JSON text
12. Look for a value called 'id', it should look something like this: `"id":"5aea17e911c71fee96375d94"`
13. Copy the idList value, in our case `5aea17e911c71fee96375d94` and enter it into the idList field
14. Lastly, lets test our flow by clicking 'Test Flow'
15. A new card should have been created in the selected list
16. Now click publish, and thats it!

### CODE REVIEW

**Comment by abuaboud** [comment]
Thank you for fixing the docs, I will take these apply them in the main branches directly, so they don't wait for this PR to be reviewed.

**Comment by ShayPunter** [comment]
Actively working on implementing the trigger now for when a new card is added, once I've added that and tested it, ill update the original PR comment to reflect testing and the change :)

**Comment by ShayPunter** [comment]
I have attempted to implement a trigger piece via Trello, though Trello doesn't have too much good documentation surrounding its webhook capabilities.

I have made a Gist for anyone who wants to take over implementing the trigger, though please do note that this hasn't been fully tested https://gist.github.com/ShayPunter/be280179628f50d4c1149c413d1edd79

Ill keep this PR open as it still implements two pieces for Trello

File: docs/_snippets/apps.mdx [hunk 1]
```diff
@@ -205,6 +205,15 @@
...
     </a>
+
+    <a href="https://activepieces.com/docs/pieces/apps/trello">
+      <Card>
+          <p align="center">
+            <strong>Trello</strong>
+            <img height="75px" width="75px" src="https://pbs.twimg.com/profile_images/1361722806694785027/UY7DlO0a_400x400.png" />
+          </p>
+      </Card>
+    </a>
```
Comment (abuaboud) [inline docs/_snippets/apps.mdx]: sorry for not mentioning this in docs, Please revert this change, It will be auto generated

File: packages/pieces/apps/src/lib/trello/index.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { createPiece } from '@activepieces/framework';
+import { createCard } from './actions/create-card';
+import { getCard } from './actions/get-card';
+
+export const trello = createPiece({
+  name: 'trello',
+  displayName: 'Trello',
+  logoUrl: 'https://pbs.twimg.com/profile_images/1361722806694785027/UY7DlO0a_400x400.png', //todo: change to trello logo on cdn
```
Comment (abuaboud) [inline packages/pieces/apps/src/lib/trello/index.ts]: Thank you, I have uploaded it `https://cdn.activepieces.com/pieces/trello.png`

File: packages/pieces/apps/src/lib/trello/index.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+export const trello = createPiece({
+  name: 'trello',
+  displayName: 'Trello',
+  logoUrl: 'https://pbs.twimg.com/profile_images/1361722806694785027/UY7DlO0a_400x400.png', //todo: change to trello logo on cdn
+  actions: [
+    createCard,
+    getCard,
+  ],
+  triggers: [],
```
Comment (abuaboud) [inline packages/pieces/apps/src/lib/trello/index.ts]: Please add your self as author, `authors: ["ShayPunter"]`

File: packages/pieces/apps/src/lib/trello/actions/create-card.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import { TrelloCard } from '../common/props/card';
+
+export const createCard = createAction({
+	name: 'create_card',
+    displayName:'Create Card',
+    description: 'Create a new card in Trello',
+	props: {
+        authentication: trelloCommon.authentication,
+        token: trelloCommon.token,
+        listId: Property.ShortText({
```
Comment (abuaboud) [inline packages/pieces/apps/src/lib/trello/actions/create-card.ts]: We can remove this step for the users, by creating dropdown that loads lists when the user connect their account.

Check this example in Google Drive. (https://github.com/activepieces/activepieces/blob/main/packages/pieces/apps/src/lib/google-drive/action/create-new-folder.ts#L26)

File: docs/pieces/apps/trello.mdx [hunk 5]
```diff
@@ -0,0 +1,63 @@
+---
```
Comment (abuaboud) [inline docs/pieces/apps/trello.mdx]: Please move this into piece folder, the docs will be generated from it.

Check discord example: https://github.com/activepieces/activepieces/blob/main/packages/pieces/apps/src/lib/discord/discord.mdx

File: packages/pieces/apps/src/lib/trello/actions/create-card.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+import { trelloCommon } from '../common';
+import { TrelloCard } from '../common/props/card';
+
+export const createCard = createAction({
+	name: 'create_card',
+    displayName:'Create Card',
+    description: 'Create a new card in Trello',
+	props: {
+        authentication: trelloCommon.authentication,
+        token: trelloCommon.token,
```
Comment (abuaboud) [inline packages/pieces/apps/src/lib/trello/actions/create-card.ts]: We can replace authentication and token, with basic authentication property It can hold two secret text.

I understand it's not exactly Basic Authentication, but you can consider api key as username and token as password for time being, check basic auth property I just updated the docs

https://github.com/activepieces/activepieces/blob/main/docs/contributing/building-pieces/reference.mdx

**Review by abuaboud — CHANGES_REQUESTED** [review]
